math/overflow_test.go

```diff
@@ -17,8 +17,10 @@ func TestAddUint64(t *testing.T) {
 	c := uint64(3)
 	sum, err = AddUint64WithErr(b, c)
 	require.Equal(t, ErrAdditionOverflow, err)
+	require.Equal(t, b+c, sum)
 
 	d := uint64(0)
 	sum, err = AddUint64WithErr(b, d)
 	require.Nil(t, err)
+	require.Equal(t, b+d, sum)
```

Comment:
Using `+` does not treat overflowing / underflowing. Please specify a value here, like `uint64(math.MaxUint64 - 2)` or whatever is expected.

Reply:
this is what the function returns.
`func AddUint64WithErr(a, b uint64) (uint64, error) {
	s := a + b
	if s >= a && s >= b {
		return s, nil
	}

	return s, ErrAdditionOverflow
}`